checks/tasks/update.py

```diff
@@ -108,28 +39,33 @@ def _update_hof():
               test scored 100%.
 
     """
-    hof = dict()
-    for model, entry_creation in ((DomainTestReport, _update_web_entry), (MailTestReport, _update_mail_entry)):
-        _populate_HOF(hof, model, entry_creation)
-
     champions = []
     web = []
     mail = []
-    for entry in hof.values():
-        is_web = False
-        is_mail = False
-        if entry.web_permalink:
-            web.append({"permalink": entry.web_permalink, "domain": entry.domain, "timestamp": entry.web_timestamp})
-            is_web = True
-        if entry.mail_permalink:
-            mail.append({"permalink": entry.mail_permalink, "domain": entry.domain, "timestamp": entry.mail_timestamp})
-            is_mail = True
-        if is_web and is_mail:
-            timestamp = entry.mail_timestamp
-            permalink = entry.mail_permalink
-            if entry.web_timestamp > entry.mail_timestamp:
-                timestamp = entry.web_timestamp
-                permalink = entry.web_permalink
+
+    for entry in Fame.objects.all().iterator():
+        if entry.site_report_id is not None:
+            web.append(
+                {
+                    "permalink": f"/site/{entry.domain}/{entry.site_report_id}/",
+                    "domain": entry.domain,
+                    "timestamp": entry.site_report_timestamp,
+                }
+            )
+        if entry.mail_report_id is not None:
+            mail.append(
+                {
+                    "permalink": f"/mail/{entry.domain}/{entry.mail_report_id}/",
+                    "domain": entry.domain,
+                    "timestamp": entry.mail_report_timestamp,
+                }
+            )
+        if entry.site_report_id is not None and entry.mail_report_id is not None:
+            timestamp = entry.mail_report_timestamp
+            permalink = f"/mail/{entry.domain}/{entry.mail_report_id}/"
+            if entry.site_report_timestamp > entry.mail_report_timestamp:
+                timestamp = entry.site_report_timestamp
+                permalink = f"/site/{entry.domain}/{entry.site_report_id}/"
             champions.append({"permalink": permalink, "domain": entry.domain, "timestamp": timestamp})
     champions = sorted(champions, key=lambda x: x["timestamp"], reverse=True)
```

Comment:
I wonder if it would make a difference in memory usage if we do the sorting at the query level so we can be using generators all the way through this function.

Reply:
Sorting needs to be done on 3 different times:
- mail
- site
- the mail and site pair

And are then stored in Redis, so it will be in memory anyway.

Reply:
As discussed: maybe we can move this to a view instead? As we won't be querying millions but only thousands of records. Combining this with Django caching decorators and nginx cache if needed for performance. This might even result in ditching the slow worker for single test altogether since the other tasks performed by slow might also be moved to other workers or rewritten to views.

- #1756